### PR TITLE
flatpak - allow to add/remove multiple flatpaks at once

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
PR #267 is stale for a long time, but the last comments in there indicate the the implementation was mostly (if not entirely) done. Picking up where it was abandoned.

The git branch used then is no longer available, so I copied the code from the PR itself. Also made a small adjustment in the `uninstall_flat()` function.

The original PR was created to handle https://github.com/ansible/ansible/issues/41996.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
flatpak

### CODE REVIEW

**Comment by russoz** [comment]
@bmillemathias to be on the safe side, and if you have the availability of course, could you please confirm the integration tests still work?

File: tests/integration/targets/flatpak/tasks/test.yml [hunk 1]
```diff
@@ -113,3 +113,135 @@
...
+      - https://flathub.org/repo/appstream/org.gnu.emacs.flatpakref
+      - org.inkscape.Inkscape
+    state: absent
+    method: "{{ method }}"
+  register: removal_result
+
+- name: Verify removal with list partially removed test result - {{ method }}
+  assert:
+    that:
+      - "removal_result.changed == true"
```
Comment (felixfontein) [inline tests/integration/targets/flatpak/tasks/test.yml:230]:
```suggestion
      - removal_result is changed
```
Similar for the other instances.

File: plugins/modules/packaging/os/flatpak.py [hunk 2]
```diff
@@ -111,10 +113,25 @@
     state: present
     remote: gnome
 
+- name: Install multiple packages
+  flatpak:
```
Comment (felixfontein) [inline plugins/modules/packaging/os/flatpak.py:117]:
```suggestion
  community.general.flatpak:
```
Same below.

**Review by felixfontein — APPROVED** [review]
I'll merge once someone confirms that the tests still pass :)

**Comment by felixfontein** [comment]
This now has a conflict. Sorry for that! On the good side, there are now tests running in CI, though :)

**Comment by felixfontein** [comment]
Ok, I think the PR is not working. There are several problems:
1. `flatpak_exists` is used wrong. It returns now two lists, but it is used as if it still returns a boolean,
2. The `_match_installed_flat_name` function which is now used in `uninstall_flat` has not been used before (no idea why it is around?).

I fixed these two issues in d9a3b9a78e6897e88b707f55dcdc85128583ba5e.

**Comment by felixfontein** [comment]
Remaining CI issues are not caused by the PR...

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
ready_for_review

**Comment by felixfontein** [comment]
recheck

**Comment by felixfontein** [comment]
@russoz can you please review this PR after my changes? I'd really like to get this finally merged, especially now that we have tests for flatpak in CI :)

**Comment by russoz** [comment]
Sure, will take a look throughout the day.

**Comment by russoz** [comment]
Looking good to me. Thanks @felixfontein for catching up where I stopped.

**Comment by felixfontein** [comment]
@jdufresne thanks for the original PR!
@russoz thanks for reviving this PR, working on it, and reviewing it!
@Andersson007 @bmillemathias thanks for reviewing and testing the original PR!

**Pull request merged**